packages/contracts/contracts/L1/L2OutputOracle.sol

```diff
@@ -42,13 +53,16 @@ contract MockL2OutputOracle {
      * @param _l2Output The L2 output of the checkpoint block.
      * @param _timestamp The L2 block timestamp that resulted in _l2Output.
      */
-    function appendL2Output(bytes32 _l2Output, uint256 _timestamp) external {
+    function appendL2Output(bytes32 _l2Output, uint256 _timestamp) external payable onlyOwner {
+        // todo: separate owner and sequencer roles
```

Comment:
If I am reading this correctly, there must be an L2 output for each L2 block? Based on the timestamp logic

Reply:
Also what can the `owner` do that the `sequencer` can't? I don't see an upgrade latch

Reply:
> If I am reading this correctly, there must be an L2 output for each L2 block? Based on the timestamp logic

I'm not seeing that, can you clarify what you mean? I think [this check](https://github.com/ethereum-optimism/optimistic-specs/blob/c2aff7c46e943c4df5f6c7a297c7980fcf0f45a6/packages/contracts/contracts/test/L2OutputOracle.t.sol#L116-L117) should cover it.

> Also what can the `owner` do that the `sequencer` can't? I don't see an upgrade latch

There's no upgrade path yet, but there will be. That said, I'll remove the todo for now, we'll track that elsewhere.

Reply:
Also confused as to what a sequencer role would mean. Figure there's only one role, which is "whitelisted output submitter". Currently that's the sequencer (as an EOA whose key will be stored in a remote signer). In the future, multiple bonded submitter.

Reply:
If your point that this role should be called "proposer"then I agree.